### DISCUSSION TITLE
clutter_id function: ML detected as clutter

### DISCUSSION THREAD

**Discussion by saranya00-s** [uobwatergroup/towerpy#79] [Q&A]
Hi,
I was using clutter id function at higher elevation. It was found that it is detecting Melting Layer as clutter. Why?

**Comment by enchilaDaSzen** [comment]
Hi @saranya00-s 

There could be a few reasons why this is happening.

- Some polarimetric signatures of the melting layer (sharp decrease in correlation coefficient $(\rho_{HV})$, high values of reflectivity ($Z_H$)) are similar to those observed in clutter. Because of this, the classifier might mistake ML pixels for clutter.
- Normalised frequency distributions (NDS) and membership functions shown in [Figure 3](https://www.sciencedirect.com/science/article/pii/S1364815223001329?via%3Dihub#fig3) of [[1]] or Figures 2-6 of [[2]] are good examples of overlapping regions for precipitation and clutter. It can be seen that the texture of polarimetric variables  can really help to smooth things out. What value are you using for the *binary_class* argument? Have you tried changing it to, say, 15? i.e. $\sigma(\rho_{HV}) = 8 + \sigma(\Phi_{DP}) = 4 + \sigma(Z_{DR}) = 2 + \sigma(Z_{H}) = 1$, this may improve the performance for your dataset.
- Also, the training datasets we used to fine-tune the classification systems came from scans taken at lower elevations, up to about 9 degrees. So, using scans from higher elevations might not be as effective. Are you using elevation angles higher than that?

Hope this helps.

[[1]] Sanchez-Rivas, D., & Rico-Ramirez, M. A. (2023). Towerpy: An open-source toolbox for processing polarimetric weather radar data. Environmental Modelling & Software, 167, 105746. https://doi.org/10.1016/j.envsoft.2023.105746
[[2]] M. A. Rico-Ramirez and I. D. Cluckie, "Classification of Ground Clutter and Anomalous Propagation Using Dual-Polarization Weather Radar," in IEEE Transactions on Geoscience and Remote Sensing, vol. 46, no. 7, pp. 1892-1904, July 2008, doi: 10.1109/TGRS.2008.916979.

[1]: https://www.sciencedirect.com/science/article/pii/S1364815223001329
[2]: https://ieeexplore.ieee.org/document/4539261
